lib/modules/manager/bazel-module/context.spec.ts

```diff
@@ -0,0 +1,112 @@
+import { Ctx } from './context';
+import { ArrayFragment, RecordFragment, StringFragment } from './fragments';
+
+describe('modules/manager/bazel-module/context', () => {
+  describe('Ctx', () => {
+    it('construct simple bazel_dep', () => {
+      const ctx = new Ctx()
+        .startRule('bazel_dep')
+        .startAttribute('name')
+        .addString('rules_foo')
+        .startAttribute('version')
+        .addString('1.2.3')
+        .endRule();
+
+      expect(ctx.results).toEqual([
+        new RecordFragment(
+          {
+            rule: new StringFragment('bazel_dep'),
+            name: new StringFragment('rules_foo'),
+            version: new StringFragment('1.2.3'),
+          },
+          true
+        ),
+      ]);
+    });
+
+    it('construct a rule with array arg', () => {
+      const ctx = new Ctx()
+        .startRule('foo_library')
+        .startAttribute('name')
+        .addString('my_library')
+        .startAttribute('srcs')
+        .startArray()
+        .addArrayItem('first')
+        .addArrayItem('second')
+        .endArray()
+        .startAttribute('tags')
+        .startRule('get_tags')
+        .endRule()
+        .endRule();
+
+      expect(ctx.results).toEqual([
+        new RecordFragment(
+          {
+            rule: new StringFragment('foo_library'),
+            name: new StringFragment('my_library'),
+            srcs: new ArrayFragment(
+              [new StringFragment('first'), new StringFragment('second')],
+              true
+            ),
+            tags: new RecordFragment(
+              { rule: new StringFragment('get_tags') },
+              true
+            ),
+          },
+          true
+        ),
+      ]);
+    });
+
+    describe('currentRecord', () => {
+      it('returns the record fragment if it is current', () => {
+        const ctx = new Ctx().startRecord();
+        expect(ctx.currentRecord).toEqual(new RecordFragment());
+      });
+
+      it('throws if it the current is not a record fragment', () => {
+        const ctx = new Ctx().startArray();
+        expect(() => ctx.currentRecord).toThrow(
+          new Error('Requested current record, but does not exist.')
+        );
+      });
+    });
+
+    describe('currentArray', () => {
+      it('returns the array fragment if it is current', () => {
+        const ctx = new Ctx().startArray();
+        expect(ctx.currentArray).toEqual(new ArrayFragment());
+      });
+
+      it('throws if it the current is not a record fragment', () => {
+        const ctx = new Ctx().startRecord();
+        expect(() => ctx.currentArray).toThrow(
+          new Error('Requested current array, but does not exist.')
+        );
+      });
+    });
+
+    it('throws if add an attribute without a record', () => {
```

Comment:
The error message code says:

> Processing an attribute but there is no parent.

But the test description says `an attribute without a record`.

Is this intended? I would expect the test description and error message to use similar words.